crates/red_knot_python_semantic/src/semantic_index/definition.rs

```diff
@@ -203,6 +214,7 @@ pub enum DefinitionKind {
     Comprehension(ComprehensionDefinitionKind),
     Parameter(AstNodeRef<ast::Parameter>),
     ParameterWithDefault(AstNodeRef<ast::ParameterWithDefault>),
+    WithItem(AstNodeRef<ast::WithItem>),
```

Comment:
Unfortunately with-statements allow for unpacking assignment:

```
>>> class MyContext:
...     def __enter__(self):
...             return 1, 2
...     def __exit__(*a, **kw):
...             pass
...
>>> with MyContext() as (a, b):
...     pass
...
>>> a
1
>>> b
2
```

This means we need to store a bit more information in `DefinitionKind::WithItem`, like we do for `DefinitionKind::Assignment` -- each symbol in the unpacking expression gets its own `Definition`, and we will need to store both the overall `WithItem` node and the specific target `Name` node in order to handle this correctly in type inference. (We don't yet handle unpacking assignment for regular assignments in inference either, but you can use the way we handle them in creating Definitions as a model.)

Let's also add a semantic-index test for an unpacking assignment from a with-item, as in the above example.

Reply:
Make sense. Thanks for explaining. I've updated the code to reflect this change in https://github.com/astral-sh/ruff/pull/12920/commits/ba8bec693640c47c882e546d6aaf0ec33fe7c097 and added a test case for unpacking assignment in https://github.com/astral-sh/ruff/pull/12920/commits/aceee2e02e048222b56b5cc70664b8f215769b6c.